### PR TITLE
chore(repo): update rust dependencies, add rust linting

### PR DESCRIPTION
**Pull request by Chriscbr**
- Update Rust dependencies
- Fix an issue where the vscode-wing package task could be executed before the vscode-wing build task
- Add rust linting with the built-in `clippy` tool to our build process (automatically runs as part of `nx run test wingc` and `nx run test wingii`)

For linting, clippy has over 600 different lints that are grouped into categories. To start, I've just enabled the "correctness" set of lints in this PR. But over time we can add more lints by simply adding directives into the source code.

Contributors using VS Code can install the `rust-analyzer` extension and set "Rust-analyzer › Check: Command" to "clippy" to start seeing the full set of lint errors/warnings in their editor.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
 profile = "default"
-channel = "1.66.0"
+channel = "1.67.1"
```
Comment (MarkMcCulloh) [inline rust-toolchain.toml:3]: Whatever fix is required for this update probably deserves its own PR (doesn't appear to be working in this draft yet)

Comment (Chriscbr) [reply rust-toolchain.toml]: Reverted

**Review by eladb — APPROVED** [review]
Nice😀
